### ISSUE TITLE
Cache doesn't work when Poetry version is upgraded.

### ISSUE THREAD

**Issue by mai-nakagawa** [actions/setup-python#494]
**Description:**
Yesterday, Poetry 1.2.0 was released. Until then, we used Poetry 1.1.15. Hence the cached content was generated by Poetry 1.1.15. After the release of 1.2.0, `poetry install` always updates the PyPI packages, however, `setup-python` does not store the updated content to the cache. Instead it ends with the following output:
```
Cache hit occurred on the primary key setup-python-Linux-python-3.8.13-poetry-c8e2c518511fd9eaf3479a28cff85a012522b925141cd747cc33a2941d20e0dd, not saving cache.
```

I assume it's because the cache key does not include the version of Poetry (e.g. [`${this.CACHE_KEY_PREFIX}-${process.env['RUNNER_OS']}-python-${this.pythonVersion}-${this.packageManager}-${hash}`](https://github.com/actions/setup-python/blob/49a521fa062d323a7f2391290b18cf928ddfe90a/src/cache-distributions/poetry-cache.ts#L58))

Workaround is to delete the cache by [Delete a GitHub Actions cache for a repository](https://docs.github.com/en/rest/actions/cache#delete-a-github-actions-cache-for-a-repository-using-a-cache-id), so that the results of Poetry 1.2.0 can be stored to cache.

**Action version:**
`actions/setup-python@v4`

**Platform:**
- [x] Ubuntu
- [x] macOS
- [x] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

**Tools version:**
I think it occurs with any python versions

**Repro steps:**
1. Write a CI step to install Poetry v1.1.15
1. Write a CI step to use `setup-python@v4` with cacheing Poetry results
1. Run the CI
1. Update the CI step to install Poetry v1.2.0
1. Re-run the CI
1. You'll see Cache doesn't work. Specifically, `poetry install` always updates the PyPI packages defined in lock file

**Expected behavior:**
Cache works. Specifically, `poetry install` returns immediately with `No dependencies to install or update`.

**Actual behavior:**
Cache doesn't work. Specifically, `poetry install` always updates the PyPI packages defined in lock file.

**Comment by vsafonkin** [comment]
Hi @mai-nakagawa, we will take a look at this, thank you for your report.

**Comment by panticmilos** [comment]
Hi @mai-nakagawa, 

Based on my [workflow](https://github.com/panticmilos/setup-python-test/runs/8193810077?check_suite_focus=true) cache is actually working. Since dependencies are not really updated if you look at the logs. Versions are updated to the same version number (that is the weird part), which can tell us the cache is being updated. There is a possibility this is a poetry issue. For this reason, I will close this issue, but if you want to continue the conversation, or reopen the issue feel free to do so :)

**Comment by mai-nakagawa** [comment]
Hi @panticmilos,

Thanks so much for your quick reproduction! Yes, it's same as what I also have seen. Also, your observation is correct - versions are updated to the same version number and the cache is *always* updated whenever you re-run the workflow.

However, I made a different conclusion when I created this GitHub issue. I thought this poetry's behavior is by design. So I thought it's a `setup-python`'s bug - it should use Poetry version number to the cache key. Then, `setup-python` would create another cache for the new Poetry version.

I think I need to check whether it's a Poetry's bug or design. I'll reopen this issue when I should find it's a Poetry's design.
